### PR TITLE
Update meta-guidelines to mention branch naming

### PR DESCRIPTION
**Pull request by kapowaz**
The README mentions branch-naming, but the meta-guidelines didn’t (yet), so this addresses that (and also refines the review steps, since we don’t permit in-situ editing).

### CODE REVIEW

**Comment by rxbynerd** [comment]
we probably shouldn't be using slashes in branch names because of how git works, we apparently had this discussion ages ago and it was decided that we shouldn't, but updates weren't made to git-whistles, so the discussion didn't actually change anything

this does look like a really good opportunity to have that discussion again, and then action it

**Comment by kapowaz** [comment]
What is the problem with using slashes in git branch names? It seems to be a pretty common thing to do — my old team used `fix/` and `feature/` as prefixes, and my GUI git client even treats them as folders:

img

File: _guidelines/00-meta.md [hunk 1]
```diff
@@ -44,18 +44,16 @@ gets updated over time.
 Coding guidelines are curated collectively.
 This is the process we choose to follow:
 
-1. Anyone can submit a pull request to change, or add to any guideline document
-   (including this one). Each document has an "edit" link at the top to help, so
-   editing is possible within Github.
+1. Create a branch named `guidelines/whatever-it-relates-to`, then submit a pull 
+   request to change or add to any existing guideline document (including this 
+   one).
```
Comment (tompave-roo) [inline _guidelines/00-meta.md]: Why have we removed this?

> Anyone can submit a pull request to change, or add to any guideline document (including this one).

In this context, it seems important.

**Comment by tompave-roo** [comment]
I agree with @rxbynerd on avoiding branch names with slashes. We've had issues with case sensitivity and branch names in the past, and the fact that git treats them as subfolders just makes it more complicated.

File: _guidelines/00-meta.md (hunk 1 above)
Comment (kapowaz) [inline _guidelines/00-meta.md]: Other than rewording it slightly, the only bit that’s actually gone is _Anyone can submit a pull request_, but I felt that was kind of implicit? If you feel strongly it should be explicit I can put it back in easily enough.

**Comment by kapowaz** [comment]
Can anyone point me to further reading on the problems with slashes in branch names? I’ve not heard about this before. Case sensitivity and branch names seems to be a separate issue, though?

File: _guidelines/00-meta.md (hunk 1 above)
Comment (tompave-roo) [inline _guidelines/00-meta.md]: I find it makes the guidelines more "accessible", especially for new starters.

**Comment by rxbynerd** [comment]
in short - branch refs are stored as files, which git can handle correctly, but because you cant have files & folders with the same name, if someone makes a branch called "guidelines" and you make a branch called "guidelines/these-are-some-guidelines", you're gonna have a bad time.

I don't have any further reading, but I imagine someone has written something on stack overflow on the topic

**Comment by kapowaz** [comment]
That does seem _potentially_ problematic, but does it actually affect you unless you literally checkout (or attempt to checkout) both branches locally? i.e. does it affect you if `guidelines` exists remotely only, and you checkout `guidelines/these-are-some-guidelines` locally? From testing it a bit there, you seem to get a warning when you try to check out either branch if it pre-exists, so with a little naming discipline it ought to be feasible to work around this fairly simply (i.e. in this repo you shouldn’t be calling your branch either `guidelines` or `blog-posts`).

**Comment by tompave-roo** [comment]
@kapowaz 

I use this function to update my local repo:

```shell
gitup() {
    git fetch --all &&
    (git checkout staging && git merge origin/staging);
    (git checkout master && git merge origin/master);
    git remote prune origin;
}
```

And when Githib gets confused with case sensitivity, this happens:

```
$ gitup
Fetching origin
From github.com:deliveroo/orderweb
 * [new branch]      cc-xxx/fix-menu-info-issues -> origin/cc-xxx/fix-menu-info-issues
Switched to branch 'staging'
Your branch is up-to-date with 'origin/staging'.
Already up-to-date.
Switched to branch 'master'
Your branch is up-to-date with 'origin/master'.
Already up-to-date.
Pruning origin
URL: git@github.com:deliveroo/orderweb.git
 * [pruned] origin/CC-XXX/fix-menu-info-issues
```
...over and over, each time I run it.

Git first things that there is a new branch on the remote, then it prunes it because it thinks it doesn't exist on the remote.

**Comment by tompave-roo** [comment]
@kapowaz @rxbynerd It's also an issue with HFS+ being case-insensitive BUT case-aware.

So if two people create the branches `FOO` and `foo`, you're gonna have a bad time.

**Comment by mezis** [comment]
> we probably shouldn't be using slashes in branch names because of how git works

I've been using slashes in branch names since early versions of Git. Never had an issue.
I get the point about `guidelines` v `guidelines/foo`, but a @tompave-roo said — that's rarely gonna happen, and if it does it's relatively easy to fix.

**Comment by kapowaz** [comment]
@tompave-roo I get where you’re coming from there; one person calls their branch `CW-123/foo` and another person calls theirs `cw-123/foo` — this does seem potentially problematic, but again easily mitigated by gently asking people to ensure their branches use lowercase only 😁 

FWIW I’ve been using a double-slash format (`cw/123/name-of-branch`) because grouping pull requests by issue name doesn’t seem particularly useful (how many different branches are you likely to have for a single JIRA issue?) whereas grouping them by team name (`cw` etc.) _does_ seem useful.

Maybe git branch naming should actually be part of the git guidelines doc? I can create an issue to track this and invite discussion about what feels good.

**Comment by mezis** [comment]
> one person calls their branch CW-123/foo and another person calls theirs cw-123/foo
> (...)
> FWIW I’ve been using a double-slash format (cw/123/name-of-branch)

Only the first of these three is "correct", because the ticket is named `CW-123` with that exact case. Unfortunately the guidelines for this have been lost at some point...

**Comment by mezis** [comment]
(I'll prep a guidelines PR for this as it's OT)

**Comment by kapowaz** [comment]
Is there any automatic integration from branch name to JIRA?

**Comment by tompave-roo** [comment]
I see.

I think this is a case where too many guidelines and rules actually make things more complicated. Even when something goes wrong, it's easily solved with a quick chat.

I guess we commented on this because you were actively recommending a slash in the guidelines, but I really don't mind.

**Comment by kapowaz** [comment]
@tompave-roo Yeah, one reason I was advocating that was to make it easier for _anyone_ to submit a pull request (since the engineering blog repo is public), and (theoretically at least) have the github integration for Slack use it to announce guideline-specific PRs. Not sure if that’s feasible yet though!

**Comment by mezis** [comment]
> Is there any automatic integration from branch name to JIRA?

There is. Or was, seems broken. Maybe because the conventionisn't followed anymore...

> I think this is a case where too many guidelines and rules actually make things more complicated

Well, the fact we need to rely on consistent naming for some integrations hints at the opposite. No consistent branch/commit naming = no integration with Jira, for one.

**Comment by mezis** [comment]
See #82.

**Comment by tompave-roo** [comment]
>> Is there any automatic integration from branch name to JIRA?
> There is. Or was, seems broken. Maybe because the conventionisn't followed anymore...

I've always consistently followed this convention:
branch: `CW-123_foo_bar` or ``CW-123-foo_bar`
commit: `CW-123 my message bla bla`

The integration worked just fine till a while ago, so it must be something else.

> No consistent branch/commit naming = no integration with Jira, for one.

Yes, fair point :+1:

**Pull request merged**